core/drivers/crypto/se050/adaptors/apis/apdu.c

```diff
@@ -524,3 +524,205 @@ sss_status_t se050_scp03_send_rotate_cmd(pSe05xSession_t ctx,
 
 	return kStatus_SSS_Fail;
 }
+
+sss_status_t se050_ecc_gen_shared_secret(pSe05xSession_t ctx, uint32_t kid,
+					 struct se050_ecc_keypub *keypub,
+					 uint8_t *secret, unsigned long *len)
+{
+	smStatus_t status = SM_OK;
+	uint8_t *pubkey = NULL;
+	uint8_t *buf = NULL;
+	size_t pubkey_len = 0;
+
+	if (!keypub || !secret || !len)
+		return kStatus_SSS_Fail;
+
+	pubkey_len = keypub->x_len + keypub->y_len + 1;
+	buf = malloc(pubkey_len);
+	if (!buf)
+		return kStatus_SSS_Fail;
+
+	*buf = 0x04;
+	pubkey = buf + 1;
+	memcpy(pubkey, keypub->x, keypub->x_len);
+	memcpy(pubkey + keypub->x_len, keypub->y, keypub->y_len);
+	status = Se05x_API_ECGenSharedSecret(ctx, kid,
+					     buf, pubkey_len, secret, len);
+	free(buf);
+	if (status != SM_OK)
+		return kStatus_SSS_Fail;
+
+	return kStatus_SSS_Success;
+}
+
+static sss_status_t set_ecc_public(Se05xSession_t *s_ctx,
+				   Se05xPolicy_t *policy,
+				   sss_se05x_object_t *k_object,
+				   SE05x_TransientType_t type,
+				   struct se050_ecc_keypub *keypub)
+{
+	size_t pubkey_len = keypub->x_len + keypub->y_len + 1;
+	smStatus_t status = SM_NOT_OK;
+	uint8_t *pubkey = NULL;
+	uint8_t *buf = NULL;
+
+	buf = malloc(pubkey_len);
+	if (!buf)
+		return kStatus_SSS_Fail;
+
+	*buf = 0x04;
+	pubkey = buf + 1;
+	memcpy(pubkey, keypub->x, keypub->x_len);
+	memcpy(pubkey + keypub->x_len, keypub->y, keypub->y_len);
+	k_object->curve_id = keypub->curve;
+	status = Se05x_API_WriteECKey(s_ctx, policy, SE05x_MaxAttemps_UNLIMITED,
+				      k_object->keyId,
+				      (SE05x_ECCurve_t)keypub->curve,
```

Comment:
Are these casts needed?

Reply:
um, only the one concerning the type.
Otherwise:
error: implicit conversion from 'SE05x_TransientType_t' to 'SE05x_INS_t' [-Werror=enum-conversion]